# try catch

## 语法

+ 语法

  ```js
  try {
    // 代码块1
  } catch(e) {
    // 代码块2
  } finally {
    // 代码块3
  }
  ```

+ 当运行 **代码1** 的时候，如果发生错误，立即停止 **代码1** 的执行，转而执行 **代码2**，错误对象为抛出的错误对象

+ **代码块3** 是什么时候都会运行

## 新语法ES2019

+ 支持我们在使用 `try catch` 错误异常处理时，选择性的给 `catch` 传入参数

+ 即我们可以不传入 `catch` 参数

  ```js
  // 正常使用 try catch
  try {
    // todo
  } catch (err){
    console.log('err：',err)
  }
  ```

  ```js
  // 在 ES10 中可以这么使用
  try {
    // todo
  } catch {
    // todo
  }
  ```
